terminus-ui/paginator/src/paginator.component.spec.ts

```diff
@@ -1,470 +1,309 @@
-      expect(component.isFirstPage(1)).toEqual(false);
-      expect(component.isFirstPage(-2)).toEqual(false);
+      expect(hostComponent.showRecordsPerPageSelector).toEqual(false);
+      expect(fixture.debugElement.query(By.css('.qa-paginator-per-page-select'))).toBeFalsy();
     });
-
   });
 
-
-  describe(`isLastPage()`, () => {
-
-    test(`should return TRUE when the page is the last in the array`, () => {
-      component.totalRecords = 20;
-      component.ngAfterViewInit();
-
-      expect(component.isLastPage(1)).toEqual(true);
-    });
-
-
-    test(`should return FALSE when the page is not the last in the array`, () => {
-      component.totalRecords = 20;
-      component.ngAfterViewInit();
-
-      expect(component.isLastPage(0)).toEqual(false);
-    });
-
-
-    test(`should return FALSE if the pagesArray doesn't exist`, () => {
-      component.pagesArray = undefined as any;
-
-      expect(component.isLastPage(1)).toEqual(false);
-    });
-
-
-    test(`should return TRUE when the page is the last in the array and not zero-based`, () => {
-      component.totalRecords = 20;
-      component.isZeroBased = false;
-      component.ngAfterViewInit();
-
-      expect(component.isLastPage(2)).toEqual(true);
-    });
+  describe(`clicking page buttons`, () => {
+
+    test(`should go to the next page`, fakeAsync(() => {
+      const fixture = createComponent(TestComponents.RecordsPerPage);
+      const hostComponent = fixture.componentInstance;
+      fixture.detectChanges();
+      const dir = 'next';
+      tick(2000);
+      clickForResults(fixture, dir);
+
+      const titleEl = fixture.debugElement.query(By.css('.qa-paginator-current-page-menu .c-button__content')).nativeElement as HTMLElement;
+      const npIndex = hostComponent.paginatorComponent.nextPageIndex;
+
+      expect(titleEl.textContent).toContain('11 - 20 of 100');
+      expect(npIndex).toEqual(1);
+    }));
+
+    test(`should go to the previous page`, fakeAsync(() => {
+      const fixture = createComponent(TestComponents.RecordsPerPage);
+      fixture.detectChanges();
+      const dir = 'previous';
+      tick(2000);
+      clickForResults(fixture, 'next');
+      clickForResults(fixture, dir);
+
+      const titleEl = fixture.debugElement.query(By.css('.qa-paginator-current-page-menu .c-button__content')).nativeElement as HTMLElement;
+
+      expect(titleEl.textContent).toContain('1 - 10 of 100');
+    }));
+
+    test(`should go to the first page and disable first & prev buttons`, fakeAsync(() => {
+      const fixture = createComponent(TestComponents.RecordsPerPage);
+      fixture.detectChanges();
+      const dir = 'first';
+      tick(2000);
+      clickForResults(fixture, 'next');
+      clickForResults(fixture, 'next');
+      clickForResults(fixture, dir);
+
+      const titleEl = fixture.debugElement.query(By.css('.qa-paginator-current-page-menu .c-button__content')).nativeElement as HTMLElement;
+      // eslint-disable-next-line max-len
```

Comment:
We should only be disabling this rule if there is truly no way around it (ie a really long string).

For items like this we can just drop it down after the `=`:

```typescript
const firstPageBut =
  fixture.debugElement.query(By.css(`.qa-paginator-${dir}-page-button .c-button`)).nativeElement as HTMLButtonElement;
```

Same note for several lines below